File: src/test/java/org/craiglist/HomePageSteps.java
package org.craiglist;

import com.google.inject.Inject;
import cucumber.api.java.en.And;
import org.craiglist.page.objects.HomePage;

import static com.codeborne.selenide.Condition.visible;

public class HomePageSteps {

    @Inject
    public HomePage homePage;

    @And("I set (english|français|español) language")
    public void setLanguage(String language) {
        homePage.languageSelector().shouldBe(visible).selectOptionContainingText(language);
    }
    @And("I open \"(.+)\" category")
    public void openCategory(String categoryName) {
        homePage.category(categoryName).click();
    }
}

File: src/test/java/org/craiglist/HomePageSteps.kt
package org.craiglist

import com.codeborne.selenide.Condition.visible
import cucumber.api.java.en.And
import org.craiglist.page.objects.HomePage

class HomePageSteps {
    var homePage = HomePage()

    @And("^I set (english|français|español) language$")
    fun setLanguage(language: String) {
        homePage.languageSelector().shouldBe(visible).selectOptionContainingText(language)
    }

    @And("I open {string} category")
    fun openCategory(categoryName: String) {
        homePage.category(categoryName).click()
    }
}
